### PR TITLE
fix the `Array#values` example

### PR DESCRIPTION
**Pull request by Alan-Liang**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -57,10 +57,10 @@ iter.return()
 To implement double-ended iterator in userland, we could use a generator with the [`function.sent` feature](https://github.com/tc39/proposal-function.sent).
 
 ```js
-Array.prototype.values = function *values(array) {
-  for (let start = 0, end = array.length; start < end;) {
-    if (function.sent === 'back') yield array[--end]
-    else yield array[start++]
+Array.prototype.values = function *values() {
+  for (let start = 0, end = this.length; start < end;) {
```
Comment (ljharb) [inline README.md:61]: this caches the length, which makes it behave differently than the spec, which does visit items added during iteration. (separately, until recently, caching the length made a loop *slower* in v8, so it's a bad coding pattern cargoculted from C++ anyways)

Comment (hax) [reply README.md:61]: It's not for cache, it's for the pointer of iteration from back. This code describe how `values` method could be upgraded from current single-ended iterator to double-ended iterator. It should not change the behavior of current usage of `value()` except someone already use `values` and send `"back"` to its `next()` method (I guess very impossible).

Comment (ljharb) [reply README.md:61]: `var a = [1]; var i = a.values(); i.next(); a.push(2); i.next()` will yield 1 and 2. This code will not.

Comment (hax) [reply README.md:61]: Oh I see. I need to investigate such edge case to see how to preserve current behavior. Thank u!

Comment (hax) [reply README.md:61]: A very rough idea is only start to save `end` after first `next('back')`. When u already consume the last item from `next('back')`, it seems reasonable to make new items can't be visited.

Comment (ljharb) [reply README.md:61]: That's not how the spec works. New items are required to be visited.

Comment (hax) [reply README.md:61]: I understand current behavior also can't visit new items if the new items are inserted before current index? Anyway I added comment to the example to explain it's just a demo, not cover the edge cases.

Comment (Alan-Liang) [reply README.md:61]: How about this?

```js
Array.prototype.values = function *values() {
  for (let start = 0, end = 0; start + end < this.length;) {
    if (function.sent === 'back') yield this[this.length - ++end]
    else yield this[start++]
  }
}
```

Comment (hax) [reply README.md:61]: yeah this should work.

**Pull request merged**